packages/lib/src/components/UPI/types.ts

```diff
@@ -2,18 +2,35 @@ import { UIElementProps } from '../types';
 
 export type UpiPaymentData = {
     paymentMethod: {
-        type: 'upi_qr' | 'upi_collect';
+        type: TX_VARIANT;
         virtualPaymentAddress?: string;
+        appId?: string;
     };
 };
 
 export enum UpiMode {
     Vpa = 'vpa',
-    QrCode = 'qrCode'
+    QrCode = 'qrCode',
+    Intent = 'intent'
 }
 
+export enum TX_VARIANT {
+    UpiCollect = 'upi_collect',
+    UpiQr = 'upi_qr',
+    UpiIntent = 'upi_intent'
+}
+
+export type App = { id: string; name: string; type?: TX_VARIANT };
+
 export interface UPIElementProps extends UIElementProps {
-    defaultMode: UpiMode;
+    defaultMode?: UpiMode;
```

Comment:
> It is better to go with Template Literal Types rather than `enum` when exposing this as configuration.
> 
> ```js
> defaultMode: 'vpa' | 'qrCode' | 'intent'
> ```
> 
> By using enums, we force merchants to import the enum itself in order to configure the Component, which is not ideal.

The only problem is we are using the `UpiMode` in multiple places, it's a bit difficult to do the code changes as i need to replace them with the string literal. 
Since we were already declaring `enum` for `UpiMode` for old upi flows anyway, i think we can keep it for v5, and i can do it for v6.